server/src/main/java/org/elasticsearch/cluster/metadata/MetadataIndexStateService.java

```diff
@@ -460,7 +505,7 @@ public ClusterState execute(final ClusterState currentState) {
                 public void clusterStateProcessed(final ClusterState oldState, final ClusterState newState) {
                     if (oldState == newState) {
                         assert blockedIndices.isEmpty() : "List of blocked indices is not empty but cluster state wasn't changed";
-                        listener.onResponse(new AddIndexBlockResponse(true, false, Collections.emptyList()));
+                        listener.onResponse(new AddIndexBlockResponse(true, false, List.of()));
```

Comment:
NIT: same here, if we're changing this line anyway we could make the empty response a constant

Reply:
aa32395